interview-questions.md

```diff
@@ -9,35 +9,182 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: 
+I am going to list all of them so that I can try to do this from memory.
 
-Researched answer:
+index, I dont think index needs params, because you are accessing all of the elements at once. 
+
+new does not require params because you are just adding the form to create a new nonexistent element, new also doesnt perform the action of creating, its just the form to allow create to do the work
+
+create also does not need params because you are providing the functionality of adding a new element. It does not exist yet so there is no need for a param. 
+
+edit requires a param because you are accessing a specific element that you are changing. 
+
+update also requires a param because like edit you are accessing a specific element to perform the update on, update also requires strong params because you want to make sure it is secure and you are giving it required inputs. 
+
+delete requires params because you are deleting a specific instance, so you need to know which one you are deleting. 
+
+
+Researched answer: I am really struggling to find any resources on this as far as (required) params, but I know you want to give create and update strong params for security of your app from malicious intent, this article says that path params are required, which obviously makes sense, if you are routing the controller around your app, the routes will also need to have the params to get there. query string params appear after a ? in the end point, followed by the params and values, order does not matter. Header parameters are typically used for authorizations. 
+source: https://www.atatus.com/blog/rest-api-best-practices-for-parameter-and-query-string-usage/
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: The first 3 that come to mind for me are generate model, generate resource, and generate migration. the first one generates a model that you store in your database that you can add change or remove the elements of like we did with blog post day, generate resource was related to API specifically, so you didn't have the view file that you did with generate model. and generate migration is used to add a migration file, example being adding columns or changing columns to your pre existing model or resource, you use migration when you need to add more or change specific data within your model/resource. 
 
-Researched answer:
+Researched answer: 5 common rails generator commands are scaffold generator which is basically just a model but it includes all of the boiler plate items. model generator which we have commonly used in the past to create our models. controller generators which are used to generate controller. migration generators are split into 2 parts, adding new columns to existing models or resources, and removing columns from existing models. 
```

Comment:
Love that you mentioned scaffold. As always, it's crucial that we are mindful of the files we are creating with a command. Nice job!